pkg/scheduler/internal/cache/cache.go

```diff
@@ -94,23 +106,88 @@ func newSchedulerCache(ttl, period time.Duration, stop <-chan struct{}) *schedul
 		period: period,
 		stop:   stop,
 
-		nodes:       make(map[string]*schedulernodeinfo.NodeInfo),
+		nodes:       make(map[string]*nodeInfoListItem),
 		nodeTree:    newNodeTree(nil),
 		assumedPods: make(map[string]bool),
 		podStates:   make(map[string]*podState),
 		imageStates: make(map[string]*imageState),
 	}
 }
 
-// Snapshot takes a snapshot of the current schedulerinternalcache. The method has performance impact,
-// and should be only used in non-critical path.
+// newNodeInfoListItem initializes a new NodeInfo.
+func newNodeInfoListItem(ni *schedulernodeinfo.NodeInfo) *nodeInfoListItem {
+	return &nodeInfoListItem{
+		info: ni,
+	}
+}
+
+// NewNodeInfoSnapshot initializes a NodeInfoSnapshot struct and returns it.
+func NewNodeInfoSnapshot() NodeInfoSnapshot {
+	return NodeInfoSnapshot{
+		NodeInfoMap: make(map[string]*schedulernodeinfo.NodeInfo),
+	}
+}
+
+// moveNodeInfoToHead moves a NodeInfo to the head of "cache.nodes" doubly
+// linked list. The head is the most recently updated NodeInfo.
+// We assume cache lock is already acquired.
+func (cache *schedulerCache) moveNodeInfoToHead(name string) {
+	ni, ok := cache.nodes[name]
+	if !ok {
+		klog.Errorf("No NodeInfo with name %v found in the cache", name)
```

Comment:
should we return here? is this covered by test?

Reply:
Added a return. I have added tests for nodes that don't exists, but the caller of this function check existence of the node in the cache and return before calling this function.